opentelemetry-sdk/src/resource/mod.rs

```diff
@@ -109,9 +141,34 @@ impl Resource {
             resource.attrs.insert(k.clone(), v.clone());
         }
 
+        if self.schema_url == other.schema_url {
+            resource.schema_url = self.schema_url.clone();
+        } else if self.schema_url.is_none() {
+            // if the other resource has schema url, use it.
+            if other.schema_url.is_some() {
+                resource.schema_url = other.schema_url.clone();
+            }
+            // else empty schema url.
+        } else {
+            // if self has schema url, use it.
+            if other.schema_url.is_none() {
+                resource.schema_url = self.schema_url.clone();
+            }
+        }
+
         resource
     }
 
+    /// Return the [schema url] of the resource. If the resource does not have a schema url, return empty string.
+    ///
+    /// [schema url]: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.9.0/specification/schemas/overview.md#schema-url
+    pub fn schema_url(&self) -> String {
```

Comment:
Would it make sense to have this be `Option<String>`? or maybe more optimally `Option<&str>`? schema_url being a valid URL or `None` may be less error prone than having it be a valid URL or an empty string.

Reply:
Sounds reasonable :+1: Updated

Reply:
Cool, could we get `&str` here? or does that cause issues